### PR TITLE
DAOS-13587 test: Re-enable skipped DAOS Seg Count test

### PR DESCRIPTION
**Pull request by arorasamarth**
Updating test to enabling CI

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: test_segcount

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
The test was skipped:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12283/2/tests

Because the skiplist has an entry against DAOS-7258. On `wolf`:
```
$ cat /scratch/CI-skip-list-master |grep seg
[['DAOS-7258', 'test_method_name', 'test_segcount']]
```


@arorasamarth The test will run if you push a commit where the title starts with DAOS-7258. I.e.
```
DAOS-7528 test: Re-enable skipped DAOS Seg Count test

Updating test to enabling CI

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: test_segcount
```

**Review by phender — CHANGES_REQUESTED** [review]
The `io/seq_count.py` test is employing an older technique for skipping tests and the https://github.com/daos-stack/daos/blob/master/src/tests/ftest/io/seg_count.py#L17 line needs to be removed along with https://github.com/daos-stack/daos/blob/master/src/tests/ftest/io/seg_count.py#L8.  Please implement this change and use the a commit message the starts with `DAOS-7258 test:` as there is still a skip-list entry for the test case as well.

**Comment by daltonbohning** [comment]
You shouldn't request gatekeeper until after
1. The tests are passing
2. The PR is "Open" instead of Draft
3. You have two reviews

File: src/tests/ftest/io/seg_count.yaml [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
+      pinned_numa_node: 1
+      nr_xs_helpers: 1
+      fabric_iface: ib1
+      fabric_iface_port: 31417
+      log_file: daos_server1.log
+      storage: auto
 pool:
-  scm_size: 60000000000
-  svcn: 3
+  size: 60gb
```
Comment (daltonbohning) [inline src/tests/ftest/io/seg_count.yaml:24]: To be generic, we can just use most available space
```suggestion
  size: 95%
```

**Comment by arorasamarth** [comment]
All tests have passed. Sending for a review.

**Review by phender — APPROVED** [review]

**Review by mjean308 — APPROVED** [review]

**Pull request merged**